druid/src/contexts.rs

```diff
@@ -242,6 +276,60 @@ impl<'a> EventCtx<'a> {
         &self.window
     }
 
+    /// Create a new window.
+    /// `T` must represent the application state provided to [`AppLauncher::launch`].
+    ///
+    /// [`AppLauncher::launch`]: struct.AppLauncher.html#method.launch
+    pub fn new_window<T: Any>(&mut self, desc: WindowDesc<T>) {
+        if self.state_types.check_window_desc::<T>() {
+            self.submit_command(
+                Command::one_shot(commands::NEW_WINDOW, desc),
+                Target::Global,
+            );
+        } else {
+            const MSG: &str = "All application windows must represent the same type of state.";
+            if cfg!(debug_assertions) {
+                panic!(MSG);
+            } else {
+                log::error!("EventCtx::new_window: {}", MSG)
+            }
+        }
+    }
+
+    /// Set the menu of the window containing the event handling widget.
+    /// `T` must represent the application state provided to [`AppLauncher::launch`].
+    ///
+    /// [`AppLauncher::launch`]: struct.AppLauncher.html#method.launch
+    pub fn set_menu<T: Any>(&mut self, menu: MenuDesc<T>) {
+        if self.state_types.check_menu_desc::<T>() {
+            self.submit_command(Command::new(commands::SET_MENU, menu), None);
```

Comment:
I know `None` defaults to the current window, but I think we should be explicit in internal code when we can. Should just use `self.window_id` instead of `None`. Same for `show_context_menu`.